db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -2982,9 +2983,10 @@ Status DBImpl::BackgroundCompaction(bool* made_progress,
       auto err_reason = versions_->io_status().ok()
                             ? BackgroundErrorReason::kCompaction
                             : BackgroundErrorReason::kManifestWrite;
-      error_handler_.SetBGError(io_s, err_reason);
+      error_handler_.SetBGError(io_s, err_reason).PermitUncheckedError();
```

Comment:
If (almost) everywhere that calls SetBGError just ignores the error, maybe the method should be changed to return a void?  Or there should be a second signature that returns a void?

Reply:
Good point. cc @anand1976

Reply:
@anand1976 Mark suggested to have 2 versions of `SetBGError`. One returns the status after we deal with the BG error, and in this case, the caller will make some actions based on the returned error. Another one is a `void` method, we just set the BG error